### PR TITLE
[Maps] Review request for Geocode to add version preview/2021-11-01-preview

### PR DESCRIPTION
**Pull request by chcmsft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+      "in": "query",
+      "description": "A string that an [ISO 3166-1 Alpha-2 region/country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). This will alter Geopolitical disputed borders and labels to align with the specified user region.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "CountryRegionSet": {
+      "name": "countryRegionSet",
+      "in": "query",
+      "description": "Comma separated string of country or region codes, e.g. FR,ES. This will limit the search to the specified countries. Only the first country code will be used for the Geocoder.",
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:114]: If only the first value will be used, why would the customer send a set of values?

Comment (pbrasil) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:114]: Also, are we following the ISO 3166-1 Alpha-2 standard here as well? If so, we should call it out.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+            "name": "postalCode",
+            "in": "query",
+            "description": "The postal code portion of an address.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "strictMatch",
+            "in": "query",
+            "description": "Restrict the geocoding result to the country or region that is specified in the countryRegion field and the state, province or territory specified in the adminDistrict field.",
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:207]: There is no countryRegion field, just countryRegionSet.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+              "strict"
+            ],
+            "x-ms-enum": {
+              "name": "StrictMatchEnum",
+              "modelAsString": true,
+              "values": [
+                {
+                  "name": "NotStrict",
+                  "value": "notstrict",
+                  "description": "Do not restrict results to the specified countryRegion and adminDistrict."
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:222]: countryRegionSet ?

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+                },
+                {
+                  "name": "Strict",
+                  "value": "strict",
+                  "description": "Restrict results to the specified countryRegion and adminDistrict."
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:227]: countryRegionSet ?

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+          "type": "array",
+          "items": {
+            "type": "number",
+            "format": "double"
+          },
+          "minItems": 4,
+          "maxItems": 4
+        },
+        "view": {
+          "description": "A string that an [ISO 3166-1 Alpha-2 region/country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). This will alter Geopolitical disputed borders and labels to align with the specified user region.",
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:407]: Seems to be missing a word. Maybe something like "A string that specifies an [...]"

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 6]
```diff
@@ -0,0 +1,686 @@
...
+          "description": "locality property",
+          "type": "string",
+          "readOnly": true
+        },
+        "neighborhood": {
+          "description": "neighborhood property",
+          "type": "string",
+          "readOnly": true
+        },
+        "adminDistricts": {
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:613]: Should we add a description like Bing Maps? Below:

The subdivision name in the country or region for an address. This element is typically treated as the first order administrative subdivision, but in some cases it is the second, third, or fourth order subdivision in a country, dependency, or region.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 7]
```diff
@@ -0,0 +1,686 @@
...
+      "items": {
+        "type": "string",
+        "enum": [
+          "Good",
+          "Ambiguous",
+          "UpHierarchy"
+        ]
+      }
+    },
+    "Confidence": {
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:667]: Please reach out to Oswaldo Ribas to get more details to add to this field.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 8]
```diff
@@ -0,0 +1,686 @@
...
+          }
+        },
+        "formattedAddress": {
+          "description": "Formatted Address property",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "MatchCodes": {
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:656]: Please reach out to Oswaldo Ribas to get more details to add to this field.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 9]
```diff
@@ -0,0 +1,686 @@
...
+          "maxItems": 2,
+          "items": {
+            "type": "number",
+            "format": "double"
+          }
+        }
+      }
+    },
+    "Bbox": {
+      "description": "information on the coordinate range for point Geometry.  Please refer to [RFC 7946](https://datatracker.ietf.org/doc/html/rfc7946#section-5) for details.",
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:327]: Capitalize first letter.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 10]
```diff
@@ -0,0 +1,686 @@
...
+          "calculationMethod": {
+            "type": "string",
+            "enum": [
+              "Interpolation",
+              "InterpolationOffset",
+              "Parcel",
+              "Rooftop"
+            ]
+          },
+          "usageTypes": {
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:361]: We know this defines if the point should be used for center of location or Routing.
Copy from BME: https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/location-data (search for usageTypes)

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json [hunk 11]
```diff
@@ -0,0 +1,686 @@
...
+    "GeocodePoints": {
+      "description": "A collection of geocode points that differ in how they were calculated and their suggested use. For a description of the points in this collection, see the Geocode Point Fields",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "geometry": {
+            "$ref": "#/definitions/GeoJSONPoint"
+          },
+          "calculationMethod": {
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:352]: We can give more details as in BME: https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/location-data (search for calculationMethod)

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 8 above)
Comment (pbrasil) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:656]: For now, definition on: https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/location-data

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 7 above)
Comment (pbrasil) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:667]: For now, definition on: https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/location-data

**Review by pbrasil — CHANGES_REQUESTED** [review]

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 1 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:114]: use multiple region here is for the future expansion. We discuss this when we define the api parameter.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 6 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:613]: this is an array field. we provide the information in its sub-field.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 10 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:362]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 11 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:352]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 9 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 5 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 3 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 2 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 7 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:670]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 8 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:658]: updated

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 6 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:615]: added

**Comment by chcmsft** [comment]
@jhendrixMSFT Can you help to review this PR?
I plan to release this on next Monday(12/6)

**Comment by jhendrixMSFT** [comment]
@iscai-msft do you know why the python track 2 build fails?  I see this in the log but it doesn't tell me much.
`
FileNotFoundError: [Errno 2] No such file or directory: 'azure/maps/search/_version.py'
`

**Comment by iscai-msft** [comment]
@jhendrixMSFT can you add a `package-version: 1.0.0b1` to the readme.python.md (or whatever version you want it to be?)

**Comment by chcmsft** [comment]
@jhendrixMSFT Has checked with our SDK team, currently search sdk is manually generated. Thus, i removed the sdk generation section in the readme.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 11 above)
Comment (pbrasil) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:352]: There is a reference to a non-existing section in the document: "[...] see the Geocode Point Fields".
Please change to "[...] see below".

**Review by pbrasil — APPROVED** [review]
Sign-off with comments.

Please make sure to fix comment called out before moving forward. I'll sync with Eva on the dates to make this public tomorrow.

File: specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json (hunk 11 above)
Comment (chcmsft) [reply specification/maps/data-plane/Search/preview/2021-11-01-preview/geocoding.json:352]: After checking the preview page, I think "For a description of the points in this collection, see the Geocode Point Fields" is redundant.
image
in the section above, client can easily see the sub fields' description.
image
in this one, client can check the detail by clicking the url.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**